src/main/java/org/jenkinsci/test/acceptance/plugins/logparser/LogParserOutputPage.java

```diff
@@ -0,0 +1,92 @@
+package org.jenkinsci.test.acceptance.plugins.logparser;
+
+import org.jenkinsci.test.acceptance.po.PageObject;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+/**
+ * The logparser plugin uses two frames.
+ *
+ * To check the content of the frames the selenium driver must be switched to the frames.
+ *
+ * @author Michael Engel.
+ */
+public class LogParserOutputPage extends PageObject {
+
+    /**
+     * Defines the different types of frames used by the parsed output of the logparser
+     */
+    public enum LogParserFrame {
+        SIDEBAR("sidebar"), CONTENT("content");
+
+        private String frameName;
+        LogParserFrame(String frameName) {
+            this.frameName = frameName;
+        }
+
+        public String getFrameName() {
+            return this.frameName;
+        }
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param po The page of the logparser.
+     */
+    public LogParserOutputPage(PageObject po) {
+        super(po.injector, po.url(""));
+    }
+
+
+    // URL to return from the opened frame to the default window
+    private String restoreURL = null;
+
+    /**
+     * Opens the defined frame in the current window.
+     *
+     * @param frame Enum of the frame to open in the current window.
+     */
+    public void openFrameInWindow(LogParserFrame frame) {
```

Comment:
What I do not like about this approach (or the design of `WebDriver` in fact) is changing the global context. Everybody who use this API needs to keep in mind to put the context back (what does it mean *back*, anyway? ` restoreWindow`, `switchToDefaultContent` or `switchToMainframe`?) otherwise some latter calls might fail with an error that makes no sense (as it will interact with something else then it should be). This includes failed assertions so the test to be written correctly will be full of `finally` blocks to put the context back.

I suggested instead the API should expose a series of command/query methods that would change the context to do its task / get the information and then put the context back in `finally` so users do not have to care it uses frame at all.